src/client/session/test.rs

```diff
@@ -247,11 +247,11 @@ async fn cluster_time_in_commands() {
         options.heartbeat_freq = Some(Duration::from_secs(1000));
         let client = EventClient::with_options(options).await;
 
-        operation(client.clone())
+        operation(client.clone_inner())
```

Comment:
Implementing `Clone` on `TestClient` broke the deref that was cloning the inner client directly here, so I added `clone_inner` to do it manually.

Reply:
I actually just ran into this myself, and I think a simpler solution would be to just update these tests to use `EventClient` instead of `Client` for the generic parameter `F`. I think they were only accidentally relying on this cloning-into-`Client` behavior anyways.